$NetBSD: patch-CMakeLists.txt,v 1.18 2026/04/13 13:31:04 adam Exp $

Look for Qt6::GuiPrivate, needed by QXslx.
Do not use application bundle on Darwin.
Do not fetch fast_float.

--- CMakeLists.txt.orig	2026-04-05 12:17:25.000000000 +0000
+++ CMakeLists.txt
@@ -26,7 +26,7 @@ SET(STELLARIUM_RELEASE_BUILD 1 CACHE BOO
 ########### Detect Qt version
 SET(ENABLE_QT6 1 CACHE BOOL "Whether to try building with Qt6. If Qt6 is not found, Qt5 will be used.")
 IF(ENABLE_QT6)
-     FIND_PACKAGE(Qt6 COMPONENTS Core QUIET)
+     FIND_PACKAGE(Qt6 COMPONENTS Core Gui GuiPrivate QUIET)
 ENDIF()
 if (NOT Qt6_FOUND)
      FIND_PACKAGE(Qt5 REQUIRED COMPONENTS Core)
@@ -252,7 +252,7 @@ IF(${CMAKE_BUILD_TYPE} MATCHES "Fuzzer")
      ENDIF()
 ENDIF()
 
-IF(APPLE)
+IF(FALSE)
      SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
      # share data location:
      SET(SDATALOC "Resources")
@@ -375,7 +375,7 @@ IF("${CMAKE_CXX_COMPILER_ID}" MATCHES "C
           SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-string-plus-int")
 	  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-string-plus-int")
      ENDIF()
-     IF(APPLE)
+     IF(FALSE)
 	  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-common -Wall -Wextra -Wno-unused-parameter -Wno-string-plus-int")
 	  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-common -Wall -Wextra -Wno-unknown-warning-option -Wno-string-plus-int")
      ENDIF()
@@ -868,7 +868,7 @@ ENDIF()
 
 ########### Set some global variables ###########
 IF(UNIX AND NOT WIN32)
-     IF(APPLE)
+     IF(FALSE)
           SET(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/Stellarium.app/Contents")
      ELSE()
           ADD_DEFINITIONS(-DINSTALL_DATADIR="${CMAKE_INSTALL_PREFIX}/${SDATALOC}")
@@ -1033,7 +1033,7 @@ IF(GENERATE_PACKAGE_TARGET)
 ENDIF()
 
 ########### macOS Bundling ###############
-IF(APPLE)
+IF(FALSE)
      MESSAGE(STATUS "macOS deployment target: ${CMAKE_OSX_DEPLOYMENT_TARGET} (${CMAKE_OSX_ARCHITECTURES})")
      SET(PACKAGE_OSX_TARGET ${CMAKE_OSX_DEPLOYMENT_TARGET})
      ADD_DEFINITIONS(-DPACKAGE_OSX_TARGET)
@@ -1096,10 +1096,7 @@ ENDIF()
 
 include("cmake/Check-from_chars.cmake")
 IF(NOT FROM_CHARS_WORKS)
-     CPMFindPackage(NAME FastFloat
-          GIT_REPOSITORY https://github.com/fastfloat/fast_float
-          VERSION 6.1.0
-          EXCLUDE_FROM_ALL yes)
+     find_package(FastFloat REQUIRED)
      GET_TARGET_PROPERTY(FastFloat_INCLUDE_DIRECTORIES
           FastFloat::fast_float INTERFACE_INCLUDE_DIRECTORIES)
      INCLUDE_DIRECTORIES(${FastFloat_INCLUDE_DIRECTORIES})
